### PR TITLE
grpc: Add gRPC logging

### PR DESCRIPTION
**Pull request by kellyma2**
This adds logging for gRPC requests in order to make debugging easier.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,6 +16,7 @@ require (
 	github.com/go-sql-driver/mysql v1.6.0
 	github.com/gorilla/handlers v1.5.1
 	github.com/gorilla/mux v1.8.0
+	github.com/grpc-ecosystem/go-grpc-middleware v1.0.0
```
Comment (sagikazarmark) [inline go.mod:19]: Can you please upgrade to the latest version?

File: cmd/dex/serve.go [hunk 2]
```diff
@@ -117,6 +117,9 @@ func runServe(options serveOptions) error {
 		return fmt.Errorf("failed to register gRPC server metrics: %v", err)
 	}
 
+	grpcLogEntry := logrus.NewEntry(logger)
```
Comment (sagikazarmark) [inline cmd/dex/serve.go:120]: I wonder if we should mark grpc logs somehow. Like adding a field called subsystem/component/channel.

@nabokihms what do you think?

**Review by sagikazarmark — APPROVED** [review]

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Oops

File: cmd/dex/serve.go (hunk 2 above)
Comment (kellyma2) [reply cmd/dex/serve.go:120]: bump

**Comment by kellyma2** [comment]
> Oops

Looks like this needs to be resolved in some way?

**Comment by sagikazarmark** [comment]
Sorry, oops was only about me accidentally approving the PR with pending comments.

**Comment by sagikazarmark** [comment]
I guess we can merge this, but we will probably drop logrus as a logger soon.

**Comment by kellyma2** [comment]
@sagikazarmark I still see '1 change requested' and it points me at your 'oops' comment.  

I'm happy to update this to not use logsrus if the alternative is already there?

**Comment by nabokihms** [comment]
We migrated to log/slog. I'm going to close this PR since it uses the logrus library. Feel free to open a new PR if you still need this feature.

**Pull request closed without merging**